python/phylanx/ast/transformation.py

```diff
@@ -418,6 +458,15 @@ def convert_to_phylanx_type(v):
         pass
     return v
 
+# Determine whether we are
+# running in a notebook or
+# not. If we are, use the
+# IORedirector.
+try:
+    get_ipython()
```

Comment:
Does `get_ipython` really check whether we're running in a notebook. Wouldn't this call succeed in any IPython environment?

Reply:
Yes, it will also succeed in an IPython environment, but not from the regular command line.

Reply:
Would this be a problem?

Reply:
It means IPython will use the IO redirection, which it doesn't need to do. I suppose I could check os.environ[_] and look for the string "jupyter" and/or "ipython". The latter feels more hacky to me.